copilot-chat.el

```diff
@@ -274,6 +274,26 @@ Argument PROMPT is the prompt to send to Copilot."
       (insert formatted-prompt))
     (copilot-chat-prompt-send)))
 
+;;;###autoload
+(defun copilot-chat-explain-symbol-at-line()
+  "Ask Copilot to explain symbol under point, given the code line as background info."
+  (interactive)
+  (unless (copilot-chat--ready-p)
+    (copilot-chat-reset))
+  (let* ((symbol (thing-at-point 'symbol))
+         (line (buffer-substring-no-properties 
+                (line-beginning-position)
+                (line-end-position)))
+         (lang (replace-regexp-in-string
+                "-mode$" ""
+                (symbol-name major-mode)))
+         (prompt (format "In %s programming language, please explain what '%s' means in the context of this code line:\n%s" 
+                        lang symbol line)))
+    (copilot-chat--prepare-buffers)
+    (with-current-buffer copilot-chat--prompt-buffer
+      (erase-buffer)
+      (insert prompt))
+    (copilot-chat-prompt-send)))
```

Comment:
Could you move this to a `(copilot-chat--explain-symbol-at-line) function like I just did for `(copilot-chat-display)` ?
So I will be able to create an advice for shell-maker.
Thank you

Reply:
And a function to create the prompt so I can also use it in shell-maker without any code redundancy.
Thank you.

Reply:
Thanks for rely. Do you mean this change? https://github.com/chep/copilot-chat.el/pull/35/files. There is advice-add and advice-remove, do you mean add those part?

Would be great to have more detail change suggestion, Thanks.

Reply:
Not sure if you mean this commit: https://github.com/chep/copilot-chat.el/pull/34/commits/2488c1eab5b74a54c90cb0b7721b574f5d167f1b

Reply:
I just need the `(copilot-chat--explain-symbol-at-line)` function to be split and the prompt to be generated by a function :

``` emacs-lisp
(defun copilot-chat--explain-symbol-prompt()
  "Returns explain symbol prompt."
  (let ((lang (replace-regexp-in-string
               "-mode$" ""
               (symbol-name major-mode))))
    (format "In %s programming language, please explain what '%s' means in the context of this code line:\n%s" 
            lang symbol line)))

(defun copilot-chat--explain-symbol-at-line(prompt)
  "Ask Copilot to explain symbol under point.
Code line is given as background info.
Frontend may override this function.
PROMPT is the prompt to send to copilot"
    (copilot-chat--prepare-buffers)
    (with-current-buffer copilot-chat--prompt-buffer
      (erase-buffer)
      (insert prompt))
    (copilot-chat-prompt-send)))

;;;###autoload
(defun copilot-chat-explain-symbol-at-line()
  "Ask Copilot to explain symbol under point.
Code line is given as background info.
  (interactive)
  (unless (copilot-chat--ready-p)
    (copilot-chat-reset))
  (let* ((symbol (thing-at-point 'symbol))
         (line (buffer-substring-no-properties 
                (line-beginning-position)
                (line-end-position)))
        (prompt (copilot-chat--explain-symbol-prompt)))
  (copilot-chat-explain-symbol-at-line prompt))
  
```

So I can create an advice for shell maker frontend, something like this:
``` emacs-lisp
(defun copilot-chat--shell-maker-explain-symbol-at-line(prompt)
    (insert-in-shell-maker buffer prompt)
  (send-shell-maker-prompt))
  
```

The idea is to put everything not frontend dependent in the interactive function and everything else in a separate one.

(I wrote the above code without any test nor compilation)

Maybe the `(copilot-chat--explain-symbol-at-line)` function could just be named `(insert-and-send-prompt)` and could be reused in many other functions but I'll do that job ;-)

Reply:
I see. That make sense to separate prompt generation and send prompt to copilot-chat. Just did that and verified it still work. Also revert the previous change. Would you mind take a look again?